vsphere/resource_vsphere_content_library.go

```diff
@@ -0,0 +1,76 @@
+package vsphere
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-vsphere/vsphere/internal/helper/contentlibrary"
+)
+
+func resourceVSphereContentLibrary() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVSphereContentLibraryCreate,
+		Delete: resourceVSphereContentLibraryDelete,
+		Read:   resourceVSphereContentLibraryRead,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The name of the content library.",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Optional description of the content library.",
+			},
+			"storage_backing": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The name of the content library.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func resourceVSphereContentLibraryRead(d *schema.ResourceData, meta interface{}) error {
+	c := meta.(*VSphereClient).restClient
+	lib, err := contentlibrary.FromName(c, d.Get("name").(string))
+	if err != nil {
+		return err
+	}
+	d.SetId(lib.ID)
+	sb := contentlibrary.FlattenStorageBackings(lib.Storage)
+	d.Set("name", lib.Name)
+	d.Set("description", lib.Description)
+	err = d.Set("storage_backing", sb)
+	if err != nil {
+		return err
```

Comment:
It would be nice if you could put some prefix in front of the error message to make tracking them in the code easier.